bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -18,47 +18,624 @@ package v1alpha1
+func (c *RKE2Config) GetConditions() clusterv1.Conditions {
+	return c.Status.Conditions
+}
+
+func (c *RKE2Config) SetConditions(conditions clusterv1.Conditions) {
+	c.Status.Conditions = conditions
+}
 
-// Rke2ConfigList contains a list of Rke2Config
-type Rke2ConfigList struct {
+// +kubebuilder:object:root=true
+
+// RKE2ConfigList contains a list of RKE2Config
+type RKE2ConfigList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
-	Items           []Rke2Config `json:"items"`
+	Items           []RKE2Config `json:"items"`
+}
+
+// DisabledItem selects a plugin Components to be disabled
+// +kubebuilder:validation:enum=rke2-coredns;rke2-ingress-nginx;rke2-metrics-server
+type DisabledItem string
+
+// CisProfile defines the CIS Benchmark profile to be activated in RKE2
+// +kubebuilder:validation:enum=cis-1.23
+type CisProfile string
+
+// Cni defines the Cni options for deploying RKE2
+// +kubebuilder:validation:enum=none;calico;canal;cilium
+type Cni string
+
+// Encoding specifies the cloud-init file encoding.
+// +kubebuilder:validation:Enum=base64;gzip;gzip+base64
+type Encoding string
+
+const (
+	// Base64 implies the contents of the file are encoded as base64.
+	Base64 Encoding = "base64"
+	// Gzip implies the contents of the file are encoded with gzip.
+	Gzip Encoding = "gzip"
+	// GzipBase64 implies the contents of the file are first base64 encoded and then gzip encoded.
+	GzipBase64 Encoding = "gzip+base64"
+)
+
+// File defines the input for generating write_files in cloud-init.
+type File struct {
+	// Path specifies the full path on disk where to store the file.
+	Path string `json:"path"`
+
+	// Owner specifies the ownership of the file, e.g. "root:root".
+	// +optional
+	Owner string `json:"owner,omitempty"`
+
+	// Permissions specifies the permissions to assign to the file, e.g. "0640".
+	// +optional
+	Permissions string `json:"permissions,omitempty"`
+
+	// Encoding specifies the encoding of the file contents.
+	// +optional
+	Encoding Encoding `json:"encoding,omitempty"`
+
+	// Content is the actual content of the file.
+	// +optional
+	Content string `json:"content,omitempty"`
+
+	// ContentFrom is a referenced source of content to populate the file.
+	// +optional
+	ContentFrom *FileSource `json:"contentFrom,omitempty"`
+}
+
+// FileSource is a union of all possible external source types for file data.
+// Only one field may be populated in any given instance. Developers adding new
+// sources of data for target systems should add them here.
+type FileSource struct {
+	// Secret represents a secret that should populate this file.
+	Secret SecretFileSource `json:"secret"`
+}
+
+// Adapts a Secret into a FileSource.
+//
+// The contents of the target Secret's Data field will be presented
+// as files using the keys in the Data field as the file names.
+type SecretFileSource struct {
+	// Name of the secret in the RKE2BootstrapConfig's namespace to use.
+	Name string `json:"name"`
```

Comment:
We can use localobjectrefence structure here, https://pkg.go.dev/k8s.io/api/core/v1#LocalObjectReference

Reply:
We could, but do we need the secret to be `local` ? We might just use a `v1.ObjectReference` instead.

Reply:
This would make it different from CAPBPK